### ISSUE TITLE
False positive for `nesting-selector-no-missing-scoping-root` when using SASS at-rules

### ISSUE THREAD

**Issue by edwin-shdw** [stylelint-scss/stylelint-config-recommended-scss#364]
When using `@mixin` the `&` should be considered valid without a "proper" scoping root as it will be resolved after using the mixin with `@include`. Only if the `@include` is used without a proper scoping root it will cause issues.

```scss
@mixin clearfix() {
    &::after {
/*  ^ will be marked as error */

        clear: both;
        content: '';
        display: block;
    }
}

.foo {
  @include clearfix; // no issues since scoped
}

@include clearfix; // will cause issues since no scope
```
